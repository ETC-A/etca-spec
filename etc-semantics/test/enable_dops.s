;-mstrict
;64,192
    movx %rx0,%exten
    movx %rx1,2
    slox %rx1,0       ; 64
    orx  %rx0,%rx1    ; set d-ops bit
    movx %exten,%rx0  ; enable d-ops
.extension dword_operations
; now we want to test that d-ops work. Simple test is to set any bits
; that don't fit in a word and look at the result. To do that here,
; we're just sliding some bits over with slod a bunch. We can't just
; set -1 as a word because `etc-as` can't encode `zext` yet.
; %r0 already has 64 (at least) in it, so...
    slod %rd0,0       ; 2^11
    slod %rd0,0       ; 2^16
    slod %rd0,0       ; 2^21
end:
    jmp end