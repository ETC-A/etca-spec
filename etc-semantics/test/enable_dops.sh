set -eu
./scripts/test.sh "$1" "$2" "test/enable_dops.bin" 64
./scripts/test.sh "$1" "$2" "test/enable_dops.bin" 192
