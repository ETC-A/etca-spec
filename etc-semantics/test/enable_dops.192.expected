<etc>
  <k>
    #halt "Halted successfully!" ~> .
  </k>
  <pc> 4294934544 </pc>
  <registers>
    0 |-> 2097152
    1 |-> 64
    2 |-> 0
    3 |-> 0
    4 |-> 0
    5 |-> 0
    6 |-> 0
    7 |-> 0
  </registers>
  <memory>
    72057594037927808 |-> b"^\x02Y\"\\ \x14\x04_\x02l\x00l\x00l\x00\x8e\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
  </memory>
  <flags>
    <carry>    false </carry>
    <zero>     false </zero>
    <negative> false </negative>
    <overflow> false </overflow>
  </flags>
  <machine-details>
    <cpuid> 192 </cpuid>
    <exten> 64 </exten>
    <reg-mode>  doubleword </reg-mode>
    <reg-width> quadword </reg-width>
    <reg-count> 8 </reg-count>
    <evil-mode> false </evil-mode>
  </machine-details>
  <in> </in>
  <out> </out>
</etc>
